src/core/hle/service/news/news.cpp

```diff
@@ -2,18 +2,836 @@
+}
+
+Result Module::SetNotificationHeader(const u32 notification_index, const NotificationHeader* header,
+                                     const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::size_t copy_size = std::min(sizeof(NotificationHeader), size);
+
+    std::memcpy(&db.notifications[notification_id], header, copy_size);
+
+    return SaveNewsDBSavedata();
+}
+
+Result Module::SetNotificationHeaderOther(const u32 notification_index,
+                                          const NotificationHeader* header,
+                                          const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::size_t copy_size = std::min(sizeof(NotificationHeader), size);
+
+    std::memcpy(&db.notifications[notification_id], header, copy_size);
+
+    return ResultSuccess;
+}
+
+Result Module::SetNotificationMessage(const u32 notification_index, const u8* message,
+                                      const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::string message_file = fmt::format("/news{:03d}.txt", notification_id);
+
+    return SaveFileToSavedata(message_file, size, message);
+}
+
+Result Module::SetNotificationImage(const u32 notification_index, const u8* image,
+                                    const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::string image_file = fmt::format("/news{:03d}.mpo", notification_id);
+
+    return SaveFileToSavedata(image_file, size, image);
+}
+
+Result Module::SaveNotification(const NotificationHeader* header, const std::size_t header_size,
+                                const u8* message, const std::size_t message_size, const u8* image,
+                                const std::size_t image_size) {
+    if (!db.header.IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    if (!header->IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    u32 notification_count = GetTotalNotifications();
+
+    // If we have reached the limit of 100 notifications, delete the oldest one
+    if (notification_count >= MAX_NOTIFICATIONS) {
+        LOG_WARNING(Service_NEWS,
+                    "Notification limit has been reached. Deleting oldest notification ID: {}",
+                    notification_ids[0]);
+        const auto result = DeleteNotification(notification_ids[0]);
+        if (result.IsError()) {
+            return result;
+        }
```

Comment:
```suggestion
        R_TRY(DeleteNotification(notification_ids[0]));
```
For these patterns and when not using ResultVal you can use the R_TRY macro to simplify them